tiledb/sm/query/writer.cc

```diff
@@ -1120,14 +1120,23 @@ Status Writer::filter_tiles(
         Status_WriterError("Incorrect number of tiles in filter_tiles"));
   }
 
+  // Reserve a vector for offsets tiles, they need to be processed after var
+  // data tiles as the processing of var data tiles depends on offset tiles.
+  std::vector<std::tuple<WriterTile*, WriterTile*, bool, bool>> args_offsets;
+  if (var_size) {
+    args_offsets.reserve(tile_num / tile_step);
+  }
+
   std::vector<std::tuple<WriterTile*, WriterTile*, bool, bool>> args;
-  args.reserve(tile_num);
+  args.reserve(tile_num - args_offsets.capacity());
 
   for (size_t tile_idx = 0; tile_idx < tile_num; tile_idx += tile_step) {
-    args.emplace_back(&(*tiles)[tile_idx], nullptr, var_size, false);
     if (var_size) {
+      args_offsets.emplace_back(&(*tiles)[tile_idx], nullptr, var_size, false);
```

Comment:
Nit : I'd like it a bit better in terms of readability if you were using `true` instead of `var_size` in `emplace_back`. Same for `false` in the else case.